### PR TITLE
Modified LoadTest to reproduce Github #157

### PR DESCRIPTION
**Pull request by Havunen**
Reproduce https://github.com/dadhi/DryIoc/issues/157

### CODE REVIEW

File: playground/LoadTest/gen/Web.Rest.API.cs [hunk 1]
```diff
@@ -10956,7 +10957,7 @@ public class PsaListReportController : ListReportController<IPsaContext>
 
     public class PsaReportFactory : ReportControllerFactory<IPsaContext>, IPsaReportFactoryService
     {
-        public static readonly PsaReportFactory Factory = null;
+        public static readonly PsaReportFactory Factory = LoadTest.Program.RootContainer.Resolve<PsaReportFactory>(ifUnresolved: IfUnresolved.Throw);
```
Comment (Havunen) [inline playground/LoadTest/gen/Web.Rest.API.cs]: This makes everything crazy.

File: playground/LoadTest/Registrations.cs [hunk 2]
```diff
@@ -470,7 +470,12 @@ private static void RegisterRestObjects(IRegistrator container)
                 container.Register<IModelConverter<WorkPriceEx>, WorkHourPriceModelConverter>(Reuse.Scoped, ifAlreadyRegistered: IfAlreadyRegistered.Replace);
                 container.Register<IModelConverter<TemporaryProjectTravelExpenseModel>, TemporaryProjectTravelsModelConverter>(Reuse.Scoped, ifAlreadyRegistered: IfAlreadyRegistered.Replace);
                 container.Register<PsaReportFactory, PsaReportFactory>(Reuse.Singleton, ifAlreadyRegistered: IfAlreadyRegistered.Throw);
-                container.RegisterDelegate<ReportControllerFactory<IPsaContext>>(c => PsaReportFactory.Factory, Reuse.Singleton, ifAlreadyRegistered: IfAlreadyRegistered.Throw);
+                container.RegisterDelegate<ReportControllerFactory<IPsaContext>>(
+                    c =>
+                    {
+                        return PsaReportFactory.Factory;
+                    }, Reuse.Singleton, ifAlreadyRegistered: IfAlreadyRegistered.Throw
```
Comment (Havunen) [inline playground/LoadTest/Registrations.cs]: Its Delegate registration but has Reuse.Singleton so it should be called only once?

**Comment by Havunen** [comment]
It does not seem to get stuck everytime but I noticed it gets stuck more easily if eveything else than `StartRandomOrderTest` is commented out.

**Comment by Havunen** [comment]
Tested with latest changes in master branch. Still not working

**Comment by Havunen** [comment]
Here is screen shot what I'm trying to demonstrate here:

image

**Comment by Havunen** [comment]
This can happen with pretty much any number of threads, but higher the count the higher the chances are.

**Comment by Havunen** [comment]
I merged latest changes from master branch.
I tested Interpreation mode and this bug can be reproduced using the new mode also.

image

**Comment by dadhi** [comment]
Good, Thank you.
I am currently doing a deep dive into decorator implementation using the ReducedLoadTest.. found some strange things. Will look at this PR later.

**Comment by dadhi** [comment]
I would merge this after we done fully with #139.

**Comment by Havunen** [comment]
I merged latest changes here to this PR. It seems like original issue cannot be reproduced anymore. 👍

**Comment by Havunen** [comment]
Blah I screwed up this branch. I will move program.cs changes to new PR...

**Pull request closed without merging**